### PR TITLE
Fix/post UI

### PR DESCRIPTION
**Pull request by Hankaji**
- Added new hooks:
  - `useDebouce`: Change value only after certain action is done for a set amount of time.
  - `useMaxScroll`: Current serve as a placeholder and has no functionality, will be implemented later.

- New component - `DropDownMenu`:
  - \<DropDownMenu />
  - \<DropDownContent />
  - \<DropDownItem />

-  UI fix:
   - Fixed Reaction on post and comments
   - Reaction now deactivate other when selected (Only 1 reaction can be active at a time)
   - Reaction deactivate itself if clicked on the active reaction
   - Reaction now function as a drop down menu instead of old one where all reactions just lying around
   - Main icon now replaced with <SmilePlus /> instead of <Heart />, <Heart /> is moved into the dropdown menu

### CODE REVIEW

File: client/src/hooks/useMaxScroll.ts
Comment (vutuanlinh2k2) [inline client/src/hooks/useMaxScroll.ts:1]: cái này k dùng thì xoá đi nè

**Review by vutuanlinh2k2 — COMMENTED** [review]
Có video demo k @Hankaji

File: client/src/hooks/useMaxScroll.ts
Comment (Hankaji) [reply client/src/hooks/useMaxScroll.ts:1]: e de do de chuan bi implement anh

Comment (Hankaji) [reply client/src/hooks/useMaxScroll.ts:1]: also video demo: 

https://github.com/user-attachments/assets/0a0a01e2-e681-4b6b-9251-962e856be4cb

**Review by vutuanlinh2k2 — APPROVED** [review]

**Pull request merged**